test/e2e/persistent_volumes.go

```diff
@@ -66,58 +66,96 @@ var _ = framework.KubeDescribe("PersistentVolumes", func() {
 			persistentVolumeTestCleanup(c, config)
 		}()
 
+		// Create the nfs server pod
 		pod := startVolumeServer(c, config)
 		serverIP := pod.Status.PodIP
 		framework.Logf("NFS server IP address: %v", serverIP)
 
+		// Define the PersistentVolume and PersistentVolumeClaim
 		pv := makePersistentVolume(serverIP)
 		pvc := makePersistentVolumeClaim(ns)
 
-		framework.Logf("Creating PersistentVolume using NFS")
+		// Create the PersistentVolume and wait for PersistentVolume.Status.Phase to be Available
+		framework.Logf("Creating PersistentVolume")
 		pv, err := c.PersistentVolumes().Create(pv)
-		Expect(err).NotTo(HaveOccurred())
+		if err != nil {
+			framework.Failf("PersistentVolume errored during creation: %v", err)
+		}
+		framework.WaitForPersistentVolumePhase(api.VolumeAvailable, c, pv.Name, 1*time.Second, 20*time.Second)
 
+		// Create the PersistentVolumeClaim
 		framework.Logf("Creating PersistentVolumeClaim")
 		pvc, err = c.PersistentVolumeClaims(ns).Create(pvc)
-		Expect(err).NotTo(HaveOccurred())
-
-		// allow the binder a chance to catch up.  should not be more than 20s.
-		framework.WaitForPersistentVolumePhase(api.VolumeBound, c, pv.Name, 1*time.Second, 30*time.Second)
+		if err != nil {
+			framework.Failf("PersistentVolumeClaim errored during creation: %v", err)
+		}
 
+		// Wait for PersistentVolume.Status.Phase to be Bound. Can take several minutes.
+		err = framework.WaitForPersistentVolumePhase(api.VolumeBound, c, pv.Name, 3*time.Second, 300*time.Second)
+		if err != nil {
+			framework.Failf("PersistentVolume did not enter a bound state: %+v", err)
+		}
+		// Check the PersistentVolume.ClaimRef.UID for non-nil value as confirmation of the bound state.
+		framework.Logf("Checking ClaimRef value of the PersistentVolume")
 		pv, err = c.PersistentVolumes().Get(pv.Name)
-		Expect(err).NotTo(HaveOccurred())
-		if pv.Spec.ClaimRef == nil {
-			framework.Failf("Expected PersistentVolume to be bound, but got nil ClaimRef: %+v", pv)
+		if pv.Spec.ClaimRef == nil || len(pv.Spec.ClaimRef.UID) == 0 {
+			pvJson, _ := json.MarshalIndent(pv, "", "  ")
+			framework.Failf("Expected PersistentVolume to be bound, but got nil ClaimRef: %+v", string(pvJson))
+		}
+
+		// Check the PersistentVolumeClaim.Status.Phase for Bound state
+		pvc, err = c.PersistentVolumeClaims(ns).Get(pvc.Name)
+		framework.Logf("Checking the bind status of the PersistentVolumeClaim: %+v", pvc.Name)
+		if pvcPhase := pvc.Status.Phase; pvcPhase != "Bound" {
+			framework.Failf("Expected PersistentVolumeClaim phase Bound.  Actual phase:  %+v.  Error: %+v", pvcPhase, err)
+		}
+
+		// Checkpod runs a command to both write to the volume and read a prewritten index.html file on the volume.
+		// If both actions succeed, exits 0.  Else exits 1
+		pvc, _ = c.PersistentVolumeClaims(ns).Get(pvc.Name)
```

Comment:
I don't *think* you need to get the pvc again. I don't think there will be a change in that obj after line 107.